src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,172 @@
+package duke;
+
+import java.util.Scanner;
+
+import java.io.IOException;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+public class Parser {
+    private static Ui ui = new Ui();
+    private static Storage storage = new Storage();
+    private static TaskList taskList = new TaskList();
+
+    public Parser() {
+
+    }
+
+    public static void parse() {
+        Scanner sc = new Scanner(System.in);
+        while (true) {
+            String command = sc.nextLine();
+            if (command.equals("bye")) {
+                sc.close();
+                ui.sayBye();
+                break;
+            } else if (command.equals("list")) {
+
+                ui.showList(TaskList.lines);
+
+            } else if (command.contains("done")) {
+                String numbers = command.substring(5);
+                try {
+                    int taskNo = Integer.parseInt(numbers);
+                } catch (NumberFormatException notANumber) {
+                    System.err.println(notANumber);
+                    ui.showNotANumberMsg();
+                    continue;
+                }
+                int taskNo = Integer.parseInt(numbers);
+                if (TaskList.lines.size() < taskNo) {
+                    ui.showOutOfBoundsMsg(TaskList.lines.size());
+                    continue;
+                }
+                if (taskNo <= 0) {
+                    ui.showLessThanZeroMsg(taskNo);
+                    continue;
+                }
+                taskNo--;
+                taskList.makeDone(storage, taskNo);
+            } else if (command.contains("delete") || command.contains("remove")) {
```

Comment:
Will using `contains` lead to some logic bugs? For example, what if the user types in the command 
`todo Delete unwanted pictures from gallery`